File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.py
# Last updated: 8/1/2025, 10:39:58 PM
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if not prices:
            return 0
        mini=prices[0]
        maxp=0
        for i in prices:
            mini=min(i,mini)
            maxp=max(maxp,i-mini)
        return maxp   

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.java
// Last updated: 8/1/2025, 10:39:51 PM
class Solution {
    public int maxProfit(int[] prices) {
        int min=Integer.MAX_VALUE;
        int max=0;
        for(int i:prices){
            min=Math.min(min,i);
            max=Math.max(max,i-min);
        }
        return max;
    }
}